src/oslogin_utils.cc

```diff
@@ -554,31 +569,35 @@ bool ParseJsonToUsers(const string& json, std::vector<string>* result) {
 }
 
 bool ParseJsonToGroups(const string& json, std::vector<Group>* result) {
-  json_object* root = NULL;
-  root = json_tokener_parse(json.c_str());
-  if (root == NULL) {
-    return false;
-  }
-
   bool ret = false;
 
-  json_object* groups = NULL;
+  json_object* root = ParseJsonRoot(json);
+  if (root == NULL) {
+    return ret;
+  }
+  json_object* groups;
+  json_type groupType;
```

Comment:
since we are here, please move `groupType` declaration closer to where it's initialized.

Reply:
Ah right, that's why this one's declared earlier:

```
oslogin_utils.cc:623:1: error: jump to label 'cleanup'
  623 | cleanup:
      | ^~~~~~~
oslogin_utils.cc:582:10: note:   from here
  582 |     goto cleanup;
      |          ^~~~~~~
oslogin_utils.cc:584:13: note:   crosses initialization of 'json_type groupType'
  584 |   json_type groupType = json_object_get_type(groups);
      |             ^~~~~~~~~
make[1]: *** [<builtin>: oslogin_utils.o] Error 1
```

I'd forgotten about that. I think I have to move this declaration back to where I had it initially.